packages/core/src/storages/request_provider.ts

```diff
@@ -568,6 +575,10 @@ export abstract class RequestProvider implements IStorage {
      * but it will never return a false positive.
      */
     async isFinished(): Promise<boolean> {
```

Comment:
I would remove implementation here as you override it in rqv1 and rqv2. It can be confusing.